addon/services/current-user.js

```diff
@@ -63,4 +63,16 @@ export default Ember.Service.extend({
             promise: this.load().catch(() => null),
         });
     }),
+
+    /**
+     * @property esPreferenceKey
+     * @return {String}
+     */
+    esPreferenceKey: Ember.computed('currentUserId', function() {
+        let currentUserId = this.get('currentUserId');
+        if (currentUserId) {
+            return encodeURIComponent(`$^${currentUserId}#@`);
```

Comment:
What is `$^...#@` doing here?

Reply:
we'd decided to use a hash of the currentUserId but I could not find a simple (native) way to do this without adding another ember addon. `encodeURIComponent` only encode certain chars such as those I added. the outcome is like so (looks kinda hashed maybe...)
`> encodeURIComponent('$^ezcos#@')
'%24%5Eezcos%23%40'`. We can definitely just return the `currentUserId` if you think it's overkill.

Reply:
If we really think a hash is necessary there are some pretty simple options. Though I'd just add a helper function instead of adding a string method. 
https://stackoverflow.com/questions/7616461/generate-a-hash-from-string-in-javascript-jquery

If you want a hash do something like that. I don't think we *need* to obfuscate the guid, but I could see reasons why it would be nice. It is a pretty simple change, so just do it, having that kind of session id could be desirable for other things as well.